dbt/include/postgres/macros/materializations/snapshot_merge.sql

```diff
@@ -2,12 +2,14 @@
 {% macro postgres__snapshot_merge_sql(target, source, insert_cols) -%}
     {%- set insert_cols_csv = insert_cols | join(', ') -%}
 
+    {%- set stcn = config.get("snapshot_table_column_names") or get_snapshot_table_column_names() -%}
```

Comment:
`get_snapshot_table_column_names` is a new macro in `dbt-adapters`. We will need to bump the pin on the lower bound of `dbt-adapters` in `dbt-postgres`.